docs/ref/dotz.md

```diff
@@ -83,33 +83,46 @@ q)"i"$0x0 vs .z.a
     When invoked via a Unix Domain Socket, it is 0.
 
 
-## `.z.ac` (HTTP auth from cookie)
+## `.z.ac` (HTTP auth)
 
 ```syntax
 .z.ac:(requestText;requestHeaderAsDictionary)
 ```
 
-Lets you define custom code to extract Single Sign On (SSO) token cookies from the HTTP header and verify it, decoding and returning the username, or instructing what action to take.
+Lets you define custom code to authorize/authenticate a HTTP request 
+e.g. inspect HTTP headers representing  oauth tokens, cookies, etc. 
+Your custom code can then return different values based on what is discovered.
 
+The function should return a 2 element list. The list of possible return values are:
+
+* User not authorized/authenticated
 ```q
-q).z.ac:{mySSOAuthenticator x[1]`Authorization}
+(0;"")
 ```
-
-where allowed return values are
-
+User not authorized. Client is sent default 401 HTTP unauthorized response.
+A HTTP callback to handle the request will not be called.
+* User authorized/authenticated
 ```q
-(0;"")              / return default 401
-(1;"username")      / authenticated username (.z.u becomes this)
-(2;"response text") / send raw response text to client
-(4;"")              / fallback to try authentication via .z.pw (V4.0 2021.07.12)
+(1;"username")
 ```
+The provided username will be used to set [`.z.u`](#zu-user-id). 
+The relevant HTTP callback to handle this request will be allowed.
+* User not authorized/authenticated (custom response)
+```q
+(2;"response text")
+```
+Customised reponse to be sent should be provided in the "response text" section.
+The "response text" should comprised of a valid HTTP response message e.g. can be used to provide a 
+401 response with a customised message.
+A HTTP callback to handle the original request will not be called.
+* Fallback to basic authentication
+```q
+(4;"")
+```
+Fallback to [basic access authentication](https://en.wikipedia.org/wiki/Basic_access_authentication#Client_side), where the username/password are base64 decoded and processed via the [`-u`](../basics/cmdline.md#-u-usr-pwd-local)/[`-U`](../basics/cmdline.md#-u-usr-pwd) file and [`.z.pw`](#zpw-validate-user) (if defined). 
+If the user is not permitted, the client will be sent a default 401 HTTP unauthorized response. Since V4.0 2021.07.12.
```

Comment:
```suggestion
The provided username is used to set [`.z.u`](#zu-user-id). 
The relevant HTTP callback to handle this request will be allowed.
* User not authorized/authenticated (custom response)
```q
(2;"response text")
```
Customised reponse to be sent should be provided in the "response text" section.
The "response text" should comprise a valid HTTP response message e.g. can be used to provide a 
401 response with a customised message.
An HTTP callback to handle the original request will not be called.
* Fallback to basic authentication
```q
(4;"")
```
Fallback to [basic access authentication](https://en.wikipedia.org/wiki/Basic_access_authentication#Client_side), where the username/password are base64 decoded and processed via the [`-u`](../basics/cmdline.md#-u-usr-pwd-local)/[`-U`](../basics/cmdline.md#-u-usr-pwd) file and [`.z.pw`](#zpw-validate-user) (if defined). 
If the user is not permitted, the client is sent a default 401 HTTP unauthorized response. Since V4.0 2021.07.12.
```